src/mutation/RelayMutationQueue.js

```diff
@@ -133,6 +133,18 @@ class RelayMutationQueue {
 
   rollback(id: ClientMutationID): void {
     const transaction = this._get(id);
+    const collisionKey = transaction.getCollisionKey();
+    const collisionQueue =
+        collisionKey && this._collisionQueueMap[collisionKey];
+    if (collisionQueue) {
+      const index = collisionQueue.indexOf(transaction);
+      if (index !== -1) {
+        collisionQueue.splice(index, 1);
+      }
+      if (collisionKey && collisionQueue.length === 0) {
```

Comment:
Checking for collisionKey here is redundant, but flow was concern that it could be null/undefined, which is not true..